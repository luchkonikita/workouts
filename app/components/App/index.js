/**
 * @flow
 */

import React, {Component} from 'react'
import {StyleSheet, NavigatorIOS} from 'react-native'
import {route as homeScreenRoute} from 'app/components/screens/HomeScreen'

export default class App extends Component {
  render () {
    return (
      <NavigatorIOS
        style={styles.Navigator}
        itemWrapperStyle={styles.NavigatorInner}
        initialRoute={homeScreenRoute} />
    )
  }
}

const styles = StyleSheet.create({
  Navigator: {
    flex: 1
  },
  NavigatorInner: {
    marginTop: 64
  }
})
