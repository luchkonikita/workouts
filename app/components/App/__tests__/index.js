/* eslint-env jest */

import 'react-native'
import React from 'react'
import App from '../'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('App', () => {
  it('renders correctly', () => {
    const output = renderer.create(<App />)
    expect(output).toMatchSnapshot()
  })

  it('renders NavigatorIOS', () => {
    const output = shallow(<App />)
    expect(output.find('NavigatorIOS')).toHaveLength(1)
  })
})
