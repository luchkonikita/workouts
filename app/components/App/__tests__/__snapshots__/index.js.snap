exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <RCTNavigator
    interactivePopGestureEnabled={undefined}
    onNavigationComplete={[Function]}
    requestedTopOfStack={0}
    style={
      Object {
        "flex": 1,
      }
    }
    vertical={undefined}>
    <RCTNavItem
      initialRoute={
        Object {
          "component": [Function],
          "rightButtonSystemIcon": "add",
          "tintColor": undefined,
          "title": "Workouts",
          "titleTextColor": undefined,
        }
      }
      rightButtonSystemIcon="add"
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "marginTop": 64,
          },
          undefined,
        ]
      }
      tintColor={undefined}
      title="Workouts"
      titleTextColor={undefined}
      translucent={true}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Home Screen
      </Text>
    </RCTNavItem>
  </RCTNavigator>
</View>
`;
